include/swift/AST/Pattern.h

```diff
@@ -290,11 +301,17 @@ class TuplePatternElt {
 };
 
 /// A pattern consisting of a tuple of patterns.
+///
+/// The tuple pattern really represents parentheses in the position of a pattern
+/// around one or more other patterns. This could be another tuple pattern, or
+/// a mixed listing of the other pattern types. The other patterns are stored in
+/// a list of TuplePatternElt's.
+/// e.g. `let (a, _) = (1, 2)`, this tuple pattern contains a NamedPattern and
+/// an AnyPattern.
```

Comment:
[I recognize you likely don't have the full context but:] This is a little bit confusing and inaccurate, especially given the bugs we've had around `ParenPattern` and `TuplePattern` (see `repairTupleOrAssociatedValuePatternIfApplicable`).

> "The tuple pattern really represents parentheses in the position of a pattern around one or more other patterns."

This should be "0 or 2 or more"; if there is only one pattern, it should be `ParenPattern`. (I made this same mistake in my earlier comment!)

For examples, I think it might be best to refer to the `Pattern` doc comment, since that's a good place to centralize information about edge cases.